File: src/commands/config/get.ts
import type { Command } from "commander";
import { notImplemented } from "../../lib/errors/cliError.js";

export class ConfigGetCommand {
  // Prints the value of one CLI configuration key.

  register(parent: Command): void {
    // Attaches `get <key>` under the `vidbyte config` group.
    parent
      .command("get <key>")
      .description("Print a CLI configuration value")
      .action(async (key: string) => this.execute(key));
  }

  private async execute(key: string): Promise<void> {
    // Will read the key from the config store and print it.
    throw notImplemented("'vidbyte config get'");
  }
}

File: src/vidbyte_cli/commands/config/get.py
"""`vidbyte-cli config get <key>` — reads a non-secret CLI setting.

It reports the *effective* value, not the stored one, so an environment override is visible
rather than mysterious. Machine output carries the provenance alongside it.
"""

from __future__ import annotations

from enum import Enum

import click

from ...lib.config import ConfigField
from ...lib.output import OutputDocument
from ...lib.runtime.context import ApplicationContext


class ConfigGetCommand:
    def register(self, parent: click.Group) -> None:
        @parent.command(name="get", help="Read an effective CLI configuration value")
        @click.argument("key", type=click.Choice([item.value for item in ConfigField]))
        @click.pass_obj
        def _run(context: ApplicationContext, key: str) -> None:
            self.execute(context, ConfigField(key))

    def execute(self, context: ApplicationContext, field: ConfigField) -> None:
        config = context.resolved_config()
        value = getattr(config, field.value)
        serialized = value.value if isinstance(value, Enum) else value
        source = config.provenance[field]
        if context.options.debug:
            context.output().diagnostic(f"{field.value} source: {source.value}")
        # Human output is the bare value so `$(vidbyte-cli config get api_url)` is usable.
        context.output().result(
            OutputDocument(
                kind="config.value",
                data={
                    "profile": config.profile,
                    "key": field.value,
                    "value": serialized,
                    "source": source.value,
                },
            ),
            str(serialized),
        )
